src/utils.ts

```diff
@@ -131,46 +131,49 @@ export const formInputCallback = (event: Event) => {
 /**
  * The global form submit callback. We need to cancel any submission
  * if a nested internals is invalid.
- * @param {Event} - The form submit event
+ * @param {HTMLFormElement} - The form element
  * @return {void}
  */
-export const formSubmitCallback = (event: Event) => {
-  /** Get the Set of elements attached to this form */
-  const form = event.target as HTMLFormElement;
-  const elements = formElementsMap.get(form);
+export const wireSubmitLogic = (form: HTMLFormElement) => {
+  const SUBMIT_BUTTON_SELECTOR = ':is(:is(button, input)[type=submit], button:not([type])):not([disabled])';
+  let submitButtonSelector = `${SUBMIT_BUTTON_SELECTOR}:not([form])`;
 
-  /**
-   * If this form does not validate then we're done
-   */
-  if (form.noValidate) {
-    return;
+  if (form.id) {
+    submitButtonSelector += `,${SUBMIT_BUTTON_SELECTOR}[form='${form.id}']`;
   }
 
-  /** If the Set has items, continue */
-  if (elements.size) {
-    const nodes = Array.from(elements);
-    /** Check the internals.checkValidity() of all nodes */
-    const validityList = nodes
-      .reverse()
-      .map(node => {
-        const internals = internalsMap.get(node);
-        return internals.reportValidity();
-      });
+  form.addEventListener('click', event => {
```

Comment:
This is nice and it does work for most use cases. But forms can be submitted with other events (like typing "Enter" in a field). So, we need to also keep in mind those use cases. Do you think opening an issue about this is worth it?

Reply:
Thx, for the annotation. However, according to the [spec](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#implicit-submission), hitting enter in a field causes "the user agent to [fire a click event](https://html.spec.whatwg.org/multipage/webappapis.html#fire-a-click-event) at that [default button](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#default-button)."

Reply:
While pressing enter in an input field, this doesn't necessarily hold true for FACE elements, FWIW. I think this is fine as is. If you do happen upon any use cases, open an issue.

Reply:
Thanks. This does works nicely!